lib/slack.ml

```diff
@@ -438,17 +438,17 @@ let generate_status_notification ?slack_user_id ?failed_steps ~(job_log : (strin
   let handler =
     match job_log with
     | [] -> None
-    | _ :: _ when is_failed_build_notification ->
+    | _ :: _ when Status_notification.is_user channel ->
```

Comment:
```ocaml
let is_failed_build_notification =
    Build.(is_failed_build n || is_canceled_build n) && (is_failed_builds_channel || Status_notification.is_user channel)
```
Why are you not checking if `Build.(is_failed_build n || is_canceled_build n)`?

Reply:
Because we decided to only send the logs to the DMs, in order not to have to much stuff in the failed builds channel, as @Khady asked. This could be a configuration option if we wanted to play with it.

Reply:
i just realized that the ~job_log value passed in (should it not be jobs_logs if it's a list?) is filtered for failed state before being passed in. I didn't know that, so i was wondering if the filtering needed to be done here.